### PR TITLE
guarantee final pipeline

### LINKED ISSUE #341: Guarantee final pipeline when assigned to marge-bot
Requirement: To guarantee one final pipeline when MR is assigned to marge-bot

Working cases:
1. When multiple MRs are batched
2. When commits exists on master with single MR (pipeline is triggered due to rebase) 

Missed case: When single MR is being handled with no commits to rebase

Helpful for running checks on approvals, compliance pipelines, CI cost saving.

### PR DESCRIPTION
**Pull request by nithyashree675**
Closes #341

### CODE REVIEW

File: marge/single_merge_job.py [hunk 1]
```diff
@@ -13,8 +13,9 @@ class SingleMergeJob(MergeJob):
     def __init__(self, *, api, user, project, repo, options, merge_request):
         super().__init__(api=api, user=user, project=project, repo=repo, options=options)
         self._merge_request = merge_request
+        self._guarantee_final_pipeline = options.guarantee_final_pipeline
 
-    def execute(self):
+    def execute(self, ):
```
Comment (snim2) [inline marge/single_merge_job.py:18]: I'm not sure this change is needed?

**Review by andreipehas — APPROVED** [review]

File: marge/single_merge_job.py [hunk 2]
```diff
@@ -61,7 +62,17 @@ def update_merge_request_and_accept(self, approvals):
                 merge_request.comment("Someone skipped the queue! Will have to try again...")
                 continue
 
-            log.info('Commit id to merge %r (into: %r)', actual_sha, target_sha)
+            if _updated_sha == actual_sha and self._options.guarantee_final_pipeline:
+                log.info('No commits on target branch to fuse, triggering pipeline...')
+                merge_request.comment("jenkins retry")
```
Comment (mattiasb) [inline marge/single_merge_job.py:67]: How does this guarantee that the final pipeline is run? It seems especially suspicious to add a comment saying "jenkins retry" here. Is that some internal smarkets CI stuff that's leaking out here? :)

Comment (abate) [reply marge/single_merge_job.py:67]: shouldn't this be a merge_request_pipeline_trigger call instead ? Maybe I'm missing something ...

Comment (nejch) [reply marge/single_merge_job.py:67]: This is essentially what was done in the community fork in this MR: https://gitlab.com/marge-org/marge-bot/-/merge_requests/387. 

This upstream project is no longer maintained as far as we can tell.

**Pull request merged**